packages/xstate-inspect/src/browser.ts

```diff
@@ -216,11 +218,12 @@ export function inspect(options?: InspectorOptions): Inspector | undefined {
     });
 
     iframe.setAttribute('src', String(url));
-  } else {
+  } else if (!targetWindow) {
     targetWindow = window.open(String(url), 'xstateinspector');
   }
 
   return {
+    targetWindow,
```

Comment:
q: why do we want to return it here?

Reply:
By returning the `targetWindow`, it will allow customers to `postMessage` to it. This is useful for custom implementations.

There are two use cases here:
1. If a user does a `window.open` locally, and passes it as an `InspectorOptions` parameter. This is useful if the user wants a "just-in-time" implementation of the XState inspector after they've already opened a window.
2. If a user specifies only the `url`. This is useful if the customer wants an eager implementation of the XState inspector.

 In the first scenario it's simply a pass-through, while in the second it's returning the newly created window.

Reply:
I've outlined behaviors in my comment below: https://github.com/statelyai/xstate/pull/3793#issuecomment-1407377281

Reply:
I think that I would prefer to remove this from the return type altogether. If you want to provide a custom `url` to a custom iframe-based inspector and you need to communicate with it - it doesn't sound too bad if you'd have to create the iframe on your own and pass that as `targetWindow`.

The problem with the current solution in this PR is that `targetWindow` is **not** available synchronously when we are made responsible for creating the iframe. So it won't always be here which makes it harder to document when you can actually rely on this property. We could expose a getter function - but then... how do we communicate to the consumer the current state of the returned value? It feels that it would complicate more than it solves - when creating the window on your own doesn't involve a lot of steps anyway (and you only do that in a single place in your app)

Reply:
The iframe is not available synchronously, but the `targetWindow` is. However I understand your comment that sometimes giving a customer too much choice can add additional complexity. If a custom cares to control the window, provide it. Otherwise, let XState do it's thing.

I can make this change as well.

Reply:
> The iframe is not available synchronously, but the targetWindow is.

It might be an illusion - IIRC it's not available synchronously, at least not the one that we are interested in. Even if you set `src` of an iframe before attaching it to the document, it "starts" in something akin to `about:blank` and "redirects" to your actual page when the main thread yields or something. Each navigation creates a new `contentWindow` - so whatever we could grab synchronously would actually be pretty useless for the consumer

Reply:
This is a good learning for me. I need to do some more deep diving reading into this! Thanks for the insights.